modal_utils/package_utils.py

```diff
@@ -19,7 +21,7 @@ def get_file_formats(module):
 BINARY_FORMATS = ["so", "S", "s", "asm"]  # TODO
 
 
-def get_module_mount_info(module_name: str) -> typing.List[typing.Tuple[bool, Path]]:
+def get_module_mount_info(module_name: str) -> typing.Sequence[typing.Tuple[bool, Path]]:
```

Comment:
This is odd - don't we *know* that the return type here is a list specifically? Not sure why pyright would get an error here.

Reply:
TIL thanks to ChatGPT. In the past, I've only used `Sequence` when I need a read-only list but the covariant aspect is new.

```
List is invariant in its type parameter, meaning that if you have a List[X], you cannot use it in place where a List[Y] is expected, even if X is a subtype of Y. This invariance is what's causing the error, as List[tuple[Literal[True], Path]] is not considered a subtype of List[Tuple[bool, Path]].

On the other hand, Sequence is covariant in its type parameter. If X is a subtype of Y, then Sequence[X] is considered a subtype of Sequence[Y]. This is why changing List to Sequence resolves the error. When you use Sequence[Tuple[bool, Path]], the function can return a sequence of tuples where the boolean can be specifically True or False, not just a generic boolean.
```

That being said I asked for a specific example that would satisfy `List[bool]` but not `List[Literal[True, False]]` and vice versa, but none given. Full text [here](https://gist.github.com/savarin/447056e690ffbd8b7e363376b2814de7).

Reply:
TIL - had no idea abou thte List vs Sequence invariant vs. covariant differences! That's cool :) Super annoying that literal `True` and `False` values are treated as subtypes rather than instances of bool though in this case... I'd consider that a bug in pyright (especially given that `mypy` approves...). Anyways, let's go with Sequence